src/sbml_jl_model.jl

```diff
@@ -1,43 +1,115 @@
+    boundary_condition = false,  # Ignore this for now
+    constant = false,  # Ignore this for now
+)
+# Todo: infected and recovered
+infected = SMBL.Species(
+    name = "I",
+    compartment = "planet_earth",
+    initial_amount =1.0,
+    substance_units="substance",
+    only_substance_units = true,  # Ignore this for now
+    boundary_condition = false,  # Ignore this for now
+    constant = false,  
+
+)
+recovery = SMBL.Species(
+    name = "R",
+    compartment = "planet_earth",
+    initial_amount =1.0,
+    substance_units="substance",
+    only_substance_units = true,  # Ignore this for now
+    boundary_condition = false,  # Ignore this for now
+    constant = false,  
+
+)
+
+# Define `Parameter`s
+alpha = SBML.Parameter(name = "alpha", value = 1e-4, constant = true)
+# Todo: beta
+beta = SBML.Parameter(name = "beta", value = 1e-3, constant = true)
+
+
+# Define Reactions
+v_infection = SBML.MathApply(
+    "*",
+    SBML.Math[
+        SBML.MathIdent("planet_earth"),
+        SBML.MathIdent("alpha"),
+        SBML.MathIdent("S"),
+        SBML.MathIdent("I"),
     ],
-    kinetic_math = KINETICMATH1,
-    reversible = false)
-REACTION2 = SBML.Reaction(reactants = [
-        SBML.SpeciesReference(species = "s1", stoichiometry = 1.0),
+)
+infected = SBML.Reaction(
+    reactants = [SBML.SpeciesReference(species = "S", stoichiometry = 1.0)],
+    products = [SBML.SpeciesReference(species = "I", stoichiometry = 1.0)],
+    kinetic_math = v_infection,
+    reversible = false,
+)
+
+# Todo: recovery
+v_recovery = SBML.MathApply(
+    "*",
+    SBML.Math[
+        SBML.MathIdent("planet_earth"),
+        SBML.MathIdent("beta"),
+        SBML.MathIdent("I"),
+        SBML.MathIdent("R"),
     ],
-    kinetic_math = KINETICMATH3,
-    reversible = true)
-PARAM1 = SBML.Parameter(name = "k1", value = 1.0, constant = true)
-MODEL1 = SBML.Model(parameters = Dict("k1" => PARAM1),
-    compartments = Dict("c1" => COMP1),
-    species = Dict("s1" => SPECIES1),
-    reactions = Dict("r1" => REACTION1))  # PL: For instance in the compartments dict, we may want to enforce that key and compartment.name are identical
-MODEL2 = SBML.Model(parameters = Dict("k1" => PARAM1),
-    compartments = Dict("c1" => COMP1),
-    species = Dict("s1" => SPECIES1),
-    reactions = Dict("r3" => REACTION2))
\ No newline at end of file
+)
+
+ recovery = SBML.Reaction(
+    reactants = [SBML.SpeciesReference(species="I",stoichiometry=1.0)],
+    products = [SBML.SpeciesReference(species="R",stoichiometry=1.0)],
+    kinetic_math =v_recovery,
+    reversible=false,
+    
+
+ )
+
+# Define `Model`
+sir_model = SBML.Model(
+    parameters = Dict("alpha" => alpha, "beta" => beta),  # Todo: beta
+    compartments = Dict("planet_earth" => dummy_compartment),
+    # Todo: species = ...
+    species = Dict(
+        "S"=>susceptible,
+        "I"=>infected,
+        "R"=>recovery,
+    )
+    #Todo:reactions =...
+    reactions =[infected,recovery]
```

Comment:
Reactions are defined slightly differently:
- [hint1](https://github.com/LCSB-BioCore/SBML.jl/blob/b9a9f341118c0bf02b697f6675fc3bc271ccbe30/src/structs.jl#L505)
- [hint2](https://github.com/SciML/SBMLToolkit.jl/blob/e13e1ed0215dea043557f62c7bd0f0a510b00177/test/systems.jl#L43)

Reply:
In the hint 2 ,what does r3 mean i don't find any r3 in the above file 

reactions = Dict(
    "infected" => infected
     "recovery => recovery 
)

how about this..

Reply:
> Reactions are defined slightly differently:
> 
> * [hint1](https://github.com/LCSB-BioCore/SBML.jl/blob/b9a9f341118c0bf02b697f6675fc3bc271ccbe30/src/structs.jl#L505)
> * [hint2](https://github.com/SciML/SBMLToolkit.jl/blob/e13e1ed0215dea043557f62c7bd0f0a510b00177/test/systems.jl#L43)

In the hint 2 ,what does r3 mean i don't find any r3 in the above file 

reactions = Dict(
    "infected" => infected
     "recovery"=> recovery 
)

how about this..

Reply:
for this iam getting error
ERROR: ParseError:
# Error @ c:\Users\VIVEK REDDY\OneDrive\Desktop\repositories\sbml_exercises\src\sbml_jl_model.jl:99:8
       "infected" => infected
       "recovery" => recovery)
#      └────────────────────┘ ── Expected `)`

Reply:
Well spotted with the "r3". I actually cannot remember why it is a dict, not a list (could be just for reasons of backward compatibility). The keys are ignored in the importer (only values are used). When we get to writing the exporter, we can come up with a reasonable convention for naming the keys. For now, you can ignore it. So what you've got is fine.

Reply:
iam not able to run it 
iam getting error near the reactions

Reply:
Ah, I think I see the error now. If I am right, then it is something you can figure out yourself and would not want to ask me about.

Reply:
😉

Reply:
![image](https://github.com/vivekreddy049/sbml_exercises/assets/134091875/9c7ed270-3dfe-4810-a646-30cad6315e64)

Reply:
Haha, yes, I am right.

Reply:
OK, little hint. The error message is a bit misleading. Just review what the syntax for dictionaries is.

Reply:
> Haha, yes, I am right.

although it doesn't show error on vs code while running it is occuring

Reply:
There is a comma missing
```julia
reactions = Dict(
"infected" => infected,  # Note the comma here
"recovery"=> recovery
)
```

Reply:
If it is not the comma, it could be that the issue comes from me replacing code with `# Todo` statements. Use one of the most simple, yet powerful debugging techniques to figure out what is going on: create an MWE, i.e. copy the code to a new file and delete everything that works, until you have the bare minimum left that reproduces the bug you are seeing.

I'll be afk for 2h.

Reply:
> If it is not the comma, it could be that the issue comes from me replacing code with `# Todo` statements. Use one of the most simple, yet powerful debugging techniques to figure out what is going on: create an MWE, i.e. copy the code to a new file and delete everything that works, until you have the bare minimum left that reproduces the bug you are seeing.
> 
> I'll be afk for 2h.

![image](https://github.com/vivekreddy049/sbml_exercises/assets/134091875/0c45ebf4-8c66-40ca-92f2-6e1a7515f58e)
@paulflang 
@TorkelE

Reply:
Unsure about the first part. But you should use ReactionSystems to create ODEProblems directly. Where did you get the ode conversion part from (if from the Catalyst docs, I would need to rewrite)?

Reply:
> Unsure about the first part. But you should use ReactionSystems to create ODEProblems directly. Where did you get the ode conversion part from (if from the Catalyst docs, I would need to rewrite)?

paul wrote it in the  first pullrequest ,checkout #1 
but the erro comes before the ode conversion,everything is interlinked from the reactions part...

Reply:
Got it, sorry, missed that.

For now I'd just avoid `structural_simplify`.